tests/perftest/TollFreeBridge.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+using BenchmarkDotNet.Attributes;
+
+using Bindings.Test;
+
+namespace PerfTest {
+
+	[Register ("StringClass")]
+	class StringClass : NSObject {
+			public override string Description => "constant";
+	}
+
+	public class TollFreeString {
+		IntPtr StringClassClassHandle = Class.GetHandle (typeof (StringClass));
+
+		string s;
+
+		StringClass sc;
+		IntPtr description;
+
+		[GlobalSetup]
+		public void ReturnString_Setup ()
+		{
+			sc = new StringClass ();
+			description = Messaging.IntPtr_objc_msgSend (sc.Handle, Selector.GetHandle ("description"));
```

Comment:
This seems a bit more complicated than necessary, can't you just create an NSString?

```csharp
public class TollFreeString {
    NSString text;
    IntPtr text_handle;

    [GlobalSetup]
    public void ReturnString_Setup ()
    {
        text = new NSString ("constant");
        text_handle = text.Handle;
    }

    [Benchmark]
    public string ReturnString_NSString ()
    {
        var d = NSString.FromHandle (text_handle);
        return s = d;
    }

    ...
```

Reply:
> can't you just create an NSString?

Yes, that would have achieved the same in this case.
But this is a subset of the tests I have, the rest is just not ready yet, and it should make more sense in a few additional commits.